core/lib/Cardano/Address/Script.hs

```diff
@@ -222,6 +227,121 @@ scriptHashFromBytes bytes
     | BS.length bytes /= credentialHashSize = Nothing
     | otherwise = Just $ ScriptHash bytes
 
+-- | Encode a 'ScriptHash' to bech32 'Text' or hex if key role is unknown.
+--  If one wants to include, valid in governance roles only, additional byte
+--  as specified in CIP-0129, the function needs to be called with withByte=true.
+--
+-- @since 4.0.0
+scriptHashToText :: ScriptHash -> KeyRole -> Bool -> Text
```

Comment:
this `Bool` parameter feels somewhat wrong, ie. [boolean blindness](https://existentialtype.wordpress.com/2011/03/15/boolean-blindness/). Would it not be better to include this extra byte into the `KeyRole`? e.g 

```
data KeyRoles = Bare KeyRole | WithByte KeyRole
```

I am not saying we should do it here, perhaps would it be useful to discuss the design first in a follow-up PR?

Reply:
I will address that in https://github.com/IntersectMBO/cardano-addresses/issues/289